libs/agno/agno/document/reader/text_reader.py

```diff
@@ -39,3 +40,54 @@ def read(self, file: Union[Path, IO[Any]]) -> List[Document]:
         except Exception as e:
             logger.error(f"Error reading: {file}: {e}")
             return []
+
+    async def async_read(self, file: Union[Path, IO[Any]]) -> List[Document]:
+        try:
+            if isinstance(file, Path):
+                if not file.exists():
+                    raise FileNotFoundError(f"Could not find file: {file}")
+
+                log_info(f"Reading asynchronously: {file}")
+                file_name = file.stem
+
+                try:
+                    import aiofiles
+
+                    async with aiofiles.open(file, "r", encoding="utf-8") as f:
+                        file_contents = await f.read()
+                except ImportError:
+                    logger.warning("aiofiles not installed, using synchronous file I/O")
+                    file_contents = file.read_text("utf-8")
+            else:
+                log_info(f"Reading uploaded file asynchronously: {file.name}")
+                file_name = file.name.split(".")[0]
+                file.seek(0)
+                file_contents = file.read().decode("utf-8")
+
+            document = Document(
+                name=file_name,
+                id=file_name,
+                content=file_contents,
+            )
+
+            if self.chunk:
+                return await self._async_chunk_document(document)
+            return [document]
+        except Exception as e:
+            logger.error(f"Error reading asynchronously: {file}: {e}")
+            return []
+
+    async def _async_chunk_document(self, document: Document) -> List[Document]:
```

Comment:
Lets add this to the base reader? This pattern should be useable by all readers no?

Reply:
Actually that is done in Priti's CSV reader PR. Lets merge that one first.